subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy

```diff
@@ -22,53 +22,49 @@ import org.gradle.profiler.BuildContext
 import org.gradle.profiler.mutations.AbstractFileChangeMutator
 import org.gradle.util.Requires
 import org.gradle.util.TestPrecondition
-import spock.lang.Unroll
 
-@Requires(TestPrecondition.LINUX)
+@Requires(TestPrecondition.NOT_WINDOWS)
 class SwiftBuildPerformanceTest extends AbstractCrossVersionPerformanceTest {
     def setup() {
         runner.minimumBaseVersion = '4.6'
         runner.targetVersions = ["6.7-20200824220048+0000"]
         runner.args += ["--parallel", "--${ParallelismBuildOptions.MaxWorkersOption.LONG_OPTION}=6"]
     }
 
-    @Unroll
-    def "up-to-date assemble on #testProject"() {
+    def "up-to-date assemble (swift)"() {
         given:
-        runner.testProject = testProject
         runner.tasksToRun = ["assemble"]
-        runner.gradleOpts = ["-Xms$maxMemory", "-Xmx$maxMemory"]
+        runner.gradleOpts = runner.projectMemoryOptions
 
         when:
         def result = runner.run()
 
         then:
         result.assertCurrentVersionHasNotRegressed()
-
-        where:
-        testProject        | maxMemory
-        'mediumSwiftMulti' | '1G'
-        'bigSwiftApp'      | '1G'
     }
 
-    @Unroll
-    def "incremental compile on #testProject"() {
+    def "incremental compile"() {
         given:
-        runner.testProject = testProject
         runner.tasksToRun = ["assemble"]
-        runner.gradleOpts = ["-Xms$maxMemory", "-Xmx$maxMemory"]
-        runner.addBuildMutator { invocationSettings -> new ChangeSwiftFileMutator(new File(invocationSettings.projectDir, fileToChange)) }
+        runner.gradleOpts = runner.projectMemoryOptions
+        runner.addBuildMutator { invocationSettings -> new ChangeSwiftFileMutator(new File(invocationSettings.projectDir, determineFileToChange(runner.testProject))) }
 
         when:
         def result = runner.run()
 
         then:
         result.assertCurrentVersionHasNotRegressed()
+    }
 
-        where:
-        testProject        | maxMemory | fileToChange
-        "mediumSwiftMulti" | '1G'      | 'lib6api3/src/main/swift/Lib6Api3Impl2Api.swift'
-        'bigSwiftApp'      | '1G'      | 'src/main/swift//AppImpl54Api3.swift'
+    static String determineFileToChange(String testProject) {
+        switch (testProject) {
+            case 'mediumSwiftMulti':
+                return 'lib6api3/src/main/swift/Lib6Api3Impl2Api.swift'
+            case 'bigSwiftApp':
+                return 'src/main/swift//AppImpl54Api3.swift'
+            default:
+                throw new IllegalArgumentException("Invalid test project ${testProject}")
+        }
     }
```

Comment:
I wonder if we could capture this in some better way, like declaring the test project as `SwiftTestProject` that has a property called `fileToChange` or something. Should `IncrementalTestProject` already have such a property BTW?

Reply:
Also, it would be nice to declaratively capture that this test requires a `SwiftTestProject` and cannot be run with anything else.

But if we model this via an `IncrementalTestProject`, can't we just have this work the same way we have with incremental Java compile? And maybe Groovy?

Reply:
Yes. We should handle this as part of https://github.com/gradle/gradle/issues/14597.